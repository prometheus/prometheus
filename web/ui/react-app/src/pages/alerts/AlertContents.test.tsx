import React from 'react';
import { shallow } from 'enzyme';
import AlertsContent from './AlertContents';

describe('AlertsContent', () => {
  const defaultProps = {
    groups: [],
    statsCount: {
      inactive: 0,
      pending: 0,
      firing: 0,
    },
  };
  const wrapper = shallow(<AlertsContent {...defaultProps} />);

  it('matches a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  [
    { selector: '#inactive-toggler', propName: 'inactive' },
    { selector: '#pending-toggler', propName: 'pending' },
    { selector: '#firing-toggler', propName: 'firing' },
  ].forEach(testCase => {
    it(`toggles the ${testCase.propName} checkbox from true to false when clicked and back to true when clicked again`, () => {
      wrapper.find(testCase.selector).invoke('onClick')(testCase.propName);
      expect(wrapper.find(testCase.selector).prop('checked')).toBe(false);
      wrapper.find(testCase.selector).invoke('onClick')(testCase.propName);
      expect(wrapper.find(testCase.selector).prop('checked')).toBe(true);
    });
  });

  it('toggles the "annotations" checkbox from false to true when clicked and back to false when clicked again', () => {
    wrapper.find('#show-annotations-toggler').invoke('onClick')();
    expect(wrapper.find('#show-annotations-toggler').prop('checked')).toBe(true);
    wrapper.find('#show-annotations-toggler').invoke('onClick')();
    expect(wrapper.find('#show-annotations-toggler').prop('checked')).toBe(false);
  });
});
