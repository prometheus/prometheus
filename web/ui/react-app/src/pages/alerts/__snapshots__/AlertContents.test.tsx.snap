// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertsContent matches a snapshot 1`] = `
<Fragment>
  <div
    className="d-flex togglers-wrapper"
  >
    <Memo(Checkbox)
      checked={true}
      id="inactive-toggler"
      key="inactive"
      onClick={[Function]}
      wrapperStyles={
        Object {
          "marginRight": 10,
        }
      }
    >
      <Badge
        className="text-capitalize"
        color="success"
        pill={false}
        tag="span"
      >
        inactive
         (
        0
        )
      </Badge>
    </Memo(Checkbox)>
    <Memo(Checkbox)
      checked={true}
      id="pending-toggler"
      key="pending"
      onClick={[Function]}
      wrapperStyles={
        Object {
          "marginRight": 10,
        }
      }
    >
      <Badge
        className="text-capitalize"
        color="warning"
        pill={false}
        tag="span"
      >
        pending
         (
        0
        )
      </Badge>
    </Memo(Checkbox)>
    <Memo(Checkbox)
      checked={true}
      id="firing-toggler"
      key="firing"
      onClick={[Function]}
      wrapperStyles={
        Object {
          "marginRight": 10,
        }
      }
    >
      <Badge
        className="text-capitalize"
        color="danger"
        pill={false}
        tag="span"
      >
        firing
         (
        0
        )
      </Badge>
    </Memo(Checkbox)>
    <Memo(Checkbox)
      checked={false}
      id="show-annotations-toggler"
      onClick={[Function]}
      wrapperStyles={
        Object {
          "marginLeft": "auto",
        }
      }
    >
      <span
        style={
          Object {
            "fontSize": "0.9rem",
            "lineHeight": 1.9,
          }
        }
      >
        Show annotations
      </span>
    </Memo(Checkbox)>
  </div>
</Fragment>
`;
